pkg/streamingpromql/aggregations/sum.go

```diff
@@ -98,20 +98,27 @@ func (g *SumAggregationGroup) accumulateHistograms(data types.InstantVectorSerie
 		}
 
 		if g.histogramSums[outputIdx] == nil {
-			data.Histograms[inputIdx].H = nil // Ensure the FloatHistogram instance is not reused when the HPoint slice is reused, as we'll retain a reference to the FloatHistogram instance below.
-
 			if lastUncopiedHistogram == p.H {
 				// We've already used this histogram for a previous point due to lookback.
 				// Make a copy of it so we don't modify the other point.
 				g.histogramSums[outputIdx] = p.H.Copy()
 				g.histogramPointCount++
+
+				// Ensure the FloatHistogram instance is not reused when the HPoint slice is reused, as we're retaining a reference to it.
+				data.Histograms[inputIdx].H = nil
```

Comment:
is this necessary? Aren't we creating a copy of `p.H` in this case so `data.Histograms[inputIdx].H` could actually be reused. (also applies to avg)

Reply:
`data.Histograms[inputIdx].H` refers to `lastUncopiedHistogram` in this case, so we must `nil` it out - otherwise something can mutate `lastUncopiedHistogram`.

Reply:
In fact, we need to do this check regardless of `g.histogramSums[outputIdx]` being `nil` - let me fix that.

Reply:
yes, but it's already nil'd out by this point right? as when we set `lastUncopiedHistogram` we call `RemoveReferencesToRetainedHistogram`

Reply:
No - `RemoveReferencesToRetainedHistogram` only looks backwards from the current point, not forwards. If we modified the later points when we call `RemoveReferencesToRetainedHistogram`, we'd remove the histograms for later points before we'd had a chance to use them.